### PR TITLE
feat: add subscription detach sample

### LINKED ISSUE #1149: Samples: please add a subscription detachment sample
Please add sample and sample test using the region tag `pubsub_detach_subscription`. See samples and tests in other languages ([Node.js](https://github.com/googleapis/nodejs-pubsub/blob/master/samples/detachSubscription.js), [Python](https://github.com/googleapis/python-pubsub/pull/152), [Java](https://github.com/googleapis/java-pubsub/pull/289)). 

@jdpedrie

### PR DESCRIPTION
**Pull request by jdpedrie**
Closes #1149.

### CODE REVIEW

**Comment by bshaffer** [comment]
Looks like it's possible if you [follow these instructions](https://github.com/googleapis/google-cloud-php/pull/3125#issuecomment-662035269)

**Comment by bshaffer** [comment]
Are we publishing docs with this feature @pradn ? If not, I'd prefer to hold off merging until the feature is GA

File: pubsub/api/test/pubsubTest.php [hunk 1]
```diff
@@ -231,6 +231,30 @@ public function testCreateAndDeletePushSubscription()
         $this->assertRegExp(sprintf('/%s/', $subscription), $output);
     }
 
+    public function testCreateAndDetachSubscription()
+    {
+        $topic = $this->requireEnv('GOOGLE_PUBSUB_TOPIC');
+        $subscription = 'test-subscription-' . rand();
```
Comment (bshaffer) [inline pubsub/api/test/pubsubTest.php:237]: Change this line to:

```php
$subscription = 'testdetachsubsxyz-' . rand();
```

File: pubsub/api/test/pubsubTest.php [hunk 2]
```diff
@@ -231,6 +231,30 @@ public function testCreateAndDeletePushSubscription()
...
+        $this->assertRegExp(sprintf('/%s/', $subscription), $output);
+
+        $output = $this->runCommand('subscription', [
+            'subscription' => $subscription,
+            '--detach' => true,
+            'project' => self::$projectId,
+        ]);
+
+        $this->assertRegExp('/Subscription detached:/', $output);
+        $this->assertRegExp(sprintf('/%s/', $subscription), $output);
```
Comment (bshaffer) [inline pubsub/api/test/pubsubTest.php:255]: May also be good to clean up the subscription after it's detached like this:

```php
        $output = $this->runCommand('subscription', [
            'subscription' => $subscription,
            '--delete' => true,
            'project' => self::$projectId,
        ]);

        $this->assertRegExp('/Subscription deleted:/', $output);
        $this->assertRegExp(sprintf('/%s/', $subscription), $output);
```

Comment (jdpedrie) [reply pubsub/api/test/pubsubTest.php:255]: Thanks for this! I meant to ask how to clean up the orphaned resource but forgot.

**Comment by anguillanneuf** [comment]
Thank you for adding the sample!

**Pull request merged**